samtranslator/model/api/api_generator.py

```diff
@@ -204,20 +206,81 @@ def _construct_stage(self, deployment, swagger):
         stage.TracingEnabled = self.tracing_enabled
 
         if swagger is not None:
-            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger)
+            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger, self.domain)
 
         if self.tags is not None:
             stage.Tags = get_tag_list(self.tags)
 
         return stage
 
+    def _construct_api_domain(self, rest_api):
+        """
+        Constructs and returns the ApiGateway Domain and BasepathMapping
+        """
+        if self.domain is None or \
+           self.domain.get('DomainName') is None or \
+           self.domain.get('CertificateArn') is None:
+            return None, None
```

Comment:
I'm wary of silent failures like this. So if I specify `DomainName` but not `CertificateArn` it'll deploy successfully, but not have my custom domain? That seems very frustrating from a customer standpoint. If certain properties are required, we should throw a 400 back to the customer so they can understand what's wrong with their template, right?

Reply:
Agreed, totally missed this. Will add the error